engines/python/setup/djl_python/transformer-neuronx.py

```diff
@@ -0,0 +1,164 @@
+            block.self_attn.to(dtype)
+            block.fc1.to(dtype)
+            block.fc2.to(dtype)
+        self.model.lm_head.to(dtype)
+        save_pretrained_split(self.model, load_path)
+        return load_path
+
+    def convert_gpt2(self, amp):
+        logging.warning("Model conversion is a slow process to do in runtime, please consider convert it"
+                        " Ahead-of-Time next time")
+        logging.info(f"Start loading the model {self.model_id_or_path}...")
+        self.model = AutoModelForCausalLM.from_pretrained(self.model_id_or_path,
+                                                          low_cpu_mem_usage=True)
+        dtype = dtypes.to_torch_dtype(amp)
+        logging.info(f"Start model conversion to INF2 format...")
+        load_path = os.path.join(tempfile.gettempdir(), self.model_id_or_path)
+        for block in self.model.transformer.h:
+            block.attn.to(dtype)
+            block.mlp.to(dtype)
+        self.model.lm_head.to(dtype)
+        self.model.save_pretrained(load_path, max_shard_size="100GB")
+        return load_path
+
+
+    def load_opt(self, amp, unroll, n_positions, need_conversion):
+        load_path = self.model_id_or_path
+        if need_conversion:
+            load_path = self.convert_opt(amp)
+        self.model = OPTForSampling.from_pretrained(load_path,
+                                               batch_size=self.batch_size,
+                                               amp=amp,
+                                               tp_degree=self.tensor_parallel_degree,
+                                               n_positions=n_positions,
+                                               unroll=unroll)
+        self.model = self.model.to_neuron()
+
+    def load_gpt2(self, amp, unroll, n_positions, need_conversion):
+        load_path = self.model_id_or_path
+        if need_conversion:
+            load_path = self.convert_gpt2(amp)
+        self.model = GPT2ForSampling.from_pretrained(load_path,
+                                                batch_size=self.batch_size,
+                                                amp=amp,
+                                                tp_degree=self.tensor_parallel_degree,
+                                                n_positions=n_positions,
+                                                unroll=unroll)
+        self.model.to_neuron()
+
+
+
+
+    def initialize(self, properties):
+        self.batch_size = int(properties.get("batch_size", 1))
+        self.tensor_parallel_degree = int(properties.get("tensor_parallel_degree", 1))
+        self.model_id_or_path = properties.get("model_id") or properties.get(
+            "model_dir")
+        dtype = properties.get("dtype", "fp32")
+        n_positions = int(properties.get("n_positions", 128))
+        unroll = properties.get("unroll", None)
+        need_conversion = properties.get("need_conversion", "true").lower() == "true"
```

Comment:
Can we detect if the file are converted or not?
Can we save a flag when we convert the file?

Reply:
that would work...